packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_multiplier.js

```diff
@@ -0,0 +1,224 @@
+            // Ensure this block is populated after initial drag from flyout.
+            if (!this.selected_multiplier) {
+                const fake_creation_event = new Blockly.Events.Create(this);
+                fake_creation_event.recordUndo = false;
+                Blockly.Events.fire(fake_creation_event);
+            }
+        }
+    },
+    updateMultiplierInput(should_use_default_value) {
+        const { contracts_for } = ApiHelpers.instance;
+
+        if (this.selected_trade_type === 'multiplier') {
+            contracts_for.getMultiplierRange(this.selected_symbol, this.selected_trade_type).then(multiplier_range => {
+                if (multiplier_range.length > 0) {
+                    const multiplier_list_dropdown = this.getField('MULTIPLIERTYPE_LIST');
+                    const multiplier_options = multiplier_range.map(o => {
+                        const option = o.toString();
+                        return [option, option];
+                    });
+
+                    multiplier_list_dropdown.updateOptions(multiplier_options, {
+                        default_value: should_use_default_value ? undefined : multiplier_list_dropdown.getValue(),
+                    });
+                }
+            });
+        } else {
+            runIrreversibleEvents(() => {
+                runGroupedEvents(false, () => {
+                    const duration_block = this.workspace.newBlock('trade_definition_tradeoptions');
+                    duration_block.initSvg();
+                    duration_block.render();
+
+                    const parent_block = this.getParent();
+                    const parent_connection = parent_block.getInput('SUBMARKET').connection;
+                    const child_connection = duration_block.previousConnection;
+                    parent_connection.connect(child_connection);
+
+                    const duration_input = duration_block.getInput('DURATION');
+
+                    const duration_shadow_block = this.workspace.newBlock('math_number_positive');
+                    duration_shadow_block.setShadow(true);
+                    duration_shadow_block.outputConnection.connect(duration_input.connection);
+                    duration_shadow_block.initSvg();
+                    duration_shadow_block.render(true);
+
+                    const stake_input = duration_block.getInput('AMOUNT');
+
+                    const stake_shadow_block = this.workspace.newBlock('math_number_positive');
+                    stake_shadow_block.setShadow(true);
+                    stake_shadow_block.setFieldValue(1, 'NUM');
+                    stake_shadow_block.outputConnection.connect(stake_input.connection);
+                    stake_shadow_block.initSvg();
+                    stake_shadow_block.render(true);
+
+                    this.dispose();
+                });
+            });
+        }
+    },
+    setCurrency: Blockly.Blocks.trade_definition_tradeoptions.setCurrency,
```

Comment:
I think copying it completely from tradeoptions is causing the creation of a duration block on every BOCK_CREATE  and END_DRAG event, lines 69-74:
```
if (
            (event.type === Blockly.Events.BLOCK_CREATE && event.ids.includes(this.id)) ||
            event.type === Blockly.Events.END_DRAG
        ) {
            this.setCurrency();
        }
```